### DISCUSSION TITLE
v3-rc1 branch error when running npm install from Elsa.Workflows.Designer project folder (No matching version found for @elsa-workflows/elsa-workflows-designer@3.0.3-preview.145)

### DISCUSSION THREAD

**Discussion by markjones-WK** [elsa-workflows/elsa-core#4589] [Q&A]
Hi I am somewhat new to Elsa 3.0. I am trying to run the Elv3-rc1 branch code and have run into a blocker.

 I have tried this locally and from the docker file in ./docker/Dockerfile.  

from the Dockerfile, when it gets to the following spot, an error is being thrown.   

# build Elsa.Workflows.Designer npm package
WORKDIR /source/src/modules/Elsa.Workflows.Designer
RUN npm install  #The error occurs here

Here is the error:
2.465 npm ERR! code ETARGET
2.467 npm ERR! notarget No matching version found for @elsa-workflows/elsa-workflows-designer@3.0.3-preview.145.
2.467 npm ERR! notarget In most cases you or one of your dependencies are requesting
2.467 npm ERR! notarget a package version that doesn't exist.

How can I get past this this that I can run the v3-rc1 code in a docker container?  

What am I missing something?

**Comment by markjones-WK** [comment]
I am trying to run the Elsa.AllInOne.Web project.  Here is the npm log file.
[2023-10-30T20_13_29_617Z-debug-0a.log](https://github.com/elsa-workflows/elsa-core/files/13209788/2023-10-30T20_13_29_617Z-debug-0a.log)

**Comment by markjones-WK** [comment]
FYI...I changed the branch to use the v3 branch as opposed to the v3-rc1 branch and it had some changes to the docker file and I was able to complete the task.  Is the v3 branch the branch to use?  The main branch appears to still have 2.0 source code.

**Reply by sfmskywalker** [reply]
Hey @markjones-WK , glad to see you're finding your way around!

Yes, the `v3` branch contains the latest and greatest **preview** version of V3. The `master` branch contains the **released** version of V2. Once 3.0 officially ships, we'll create a `2.x` branch off of `master` and `v3` will be merged into `master`.

**Reply by markjones-WK** [reply]
Thanks!  Do you know when the planned v3 release date will be?

**Reply by sfmskywalker** [reply]
Yes, we're aiming for December 1, if not, before the end of the year.
